netutil/netutil.go

```diff
@@ -0,0 +1,129 @@
+package netutil
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+func ipForAddr(addr net.Addr) (net.IP, bool) {
+	switch a := addr.(type) {
+	case *net.IPAddr:
+		return a.IP, true
+	case *net.IPNet:
+		return a.IP, true
```

Comment:
If `addr` is a network address I'm not sure it's a good idea to just return the address of the network without any indication that this isn't a host address. IIUC the users of `PrivateNetworkInterfaces` rely on that the returned addresses are unique in the network, which I believe they won't be if we also return network addresses.